From f58b53f504d9ddd91294ab644e502de925d920d2 Mon Sep 17 00:00:00 2001
From: David Steele <dsteele@gmail.com>
Date: Sun, 28 Apr 2013 20:53:28 -0400
Subject: [PATCH 03/20] Add an 'authenticate only' option that logs in, then
 exits.

This facilitates implementation of an init.d service on top.
---
 cloudprint/cloudprint.py | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/cloudprint/cloudprint.py b/cloudprint/cloudprint.py
index 6a3dacc..8c8c4b5 100755
--- a/cloudprint/cloudprint.py
+++ b/cloudprint/cloudprint.py
@@ -381,14 +381,16 @@ def usage():
     print '-a account_file\t: path to google account ident data (default ~/.cloudprintauth)'
     print '\t\t account_file format:\t <Google username>'
     print '\t\t\t\t\t <Google password>'
+    print '-c\t\t: establish and store login credentials, then exit'
     print '-h\t\t: display this help'
 
 def main():
-    opts, args = getopt.getopt(sys.argv[1:], 'dlhp:a:')
+    opts, args = getopt.getopt(sys.argv[1:], 'dlhp:a:c')
     daemon = False
     logout = False
     pidfile = None
     authfile = None
+    authonly = False
     saslauthfile = None
     for o, a in opts:
         if o == '-d':
@@ -400,6 +402,8 @@ def main():
         elif o == '-a':
             authfile = a
             saslauthfile = authfile+'.sasl'
+        elif o == '-c':
+            authonly = True
         elif o =='-h':
             usage()
             sys.exit()
@@ -453,6 +457,9 @@ def main():
             #reset the stored auth
             cpp.set_auth('')
 
+    if authonly:
+        sys.exit(0)
+
     printers = cpp.get_printers()
 
     if daemon:
-- 
1.8.4.rc3

